extensions/ql-vscode/src/local-queries/local-query-run.ts

```diff
@@ -92,6 +97,10 @@ export class LocalQueryRun {
     // Note we must update the query history view after showing results as the
     // display and sorting might depend on the number of results
     await this.queryHistoryManager.refreshTreeView();
+
+    if (isDisposable(this.logger)) {
```

Comment:
I think the disposal of the logger is currently being done in all the right places. I was wondering whether it would be clearer and safer to make `LocalQueryRun` implement `Disposable`. I decided that I don't think it matters much. It would make some things more consistent but we'd hit a blocker in `debugger-ui.ts` and ultimately I'm not sure it brings any extra safety.

So overall I don't think anything needs to change. I just wanted to post my thoughts on it 🤷🏼